src/__tests__/ProductListItem.test.js

```diff
@@ -0,0 +1,50 @@
+import { render, fireEvent } from "@testing-library/vue"
+
+import ProductListItem from "../components/ProductListItem/ProductListItem.vue"
+
+describe("Test ProductListItemComponent", () => {
+  const item = {
+    id: 1,
+    favorite: false,
+    title: "Item Name",
+    price: 120,
+  }
+
+  function r() {
+    return render(ProductListItem, {
+      props: { item },
+    })
+  }
+
+  it("renders correctly with provided props", () => {
+    const { getByText } = r()
+    expect(getByText(item.title)).toBeInTheDocument()
+  })
+
+  it("emits edit-item event on edit button click", async () => {
+    const { getByRole, emitted } = r()
+
+    const editBtn = getByRole("button", { name: /edit/i })
+    await fireEvent.click(editBtn)
+    // console.log(emitted("edit-item"))
+    expect(emitted("edit-item")).toHaveProperty("edit-item")
```

Comment:
Try checking whether your event was called with the expected payload.
Try something like `expect(emitted('edit-item')[0][0]).toEqual(your expected object details)`